cognite/neat/graph/loaders/_rdf2dms.py

```diff
@@ -0,0 +1,191 @@
+        self.add_class_prefix = add_class_prefix
+
+    @classmethod
+    def from_data_model_id(
+        cls,
+        client: CogniteClient,
+        data_model_id: dm.DataModelId,
+        graph_store: NeatGraphStoreBase,
+        add_class_prefix: bool = False,
+    ) -> "DMSLoader":
+        # Todo add error handling
+        data_model = client.data_modeling.data_models.retrieve(data_model_id, inline_views=True).latest_version()
+        return cls(graph_store, data_model, {}, add_class_prefix)
+
+    @classmethod
+    def from_rules(
+        cls, rules: DMSRules, graph_store: NeatGraphStoreBase, add_class_prefix: bool = False
+    ) -> "DMSLoader":
+        schema = rules.as_schema()
+        # Todo add error handling
+        return cls(graph_store, schema.as_read_model(), {}, add_class_prefix)
+
+    def _load(self, stop_on_exception: bool = False) -> Iterable[dm.InstanceApply | NeatValidationError]:
+        for view in self.data_model.views:
+            view_id = view.as_id()
+            # Todo Some tracking and creation of a structure to do validation
+            validation_structure = self._create_validation_structure(view)  # type: ignore[var-annotated]
+            uri_ref = self.class_by_view_id.get(view.as_id(), URIRef(f"{view.space}:{view.external_id}"))
+            triples = self.graph_store.queries.list_instances_of_type(uri_ref)
+            for identifier, properties in _triples2dictionary(triples).items():
+                try:
+                    yield self._create_instance(identifier, properties, validation_structure, view_id)
+                except Exception:
+                    # Todo Convert to NeatValidationError
+                    raise
+
+    def load_into_cdf_iterable(self, client: CogniteClient, dry_run: bool = False) -> Iterable:
+        raise NotImplementedError()
+
+    def write_to_file(self, filepath: Path) -> None:
+        if filepath.suffix not in [".json", ".yaml", ".yml"]:
+            raise ValueError(f"File format {filepath.suffix} is not supported")
+        dumped: dict[str, list] = {"nodes": [], "edges": [], "errors": []}
+        for item in self.load(stop_on_exception=False):
+            key = {
+                dm.NodeApply: "nodes",
+                dm.EdgeApply: "edges",
+                NeatValidationError: "errors",
+            }.get(type(item))
+            if key is None:
+                # Todo use appropriate warning
+                warnings.warn(f"Item {item} is not supported", UserWarning, stacklevel=2)
+                continue
+            dumped[key].append(item.dump())
+        with filepath.open("w", encoding=self._encoding, newline=self._new_line) as f:
+            if filepath.suffix == ".json":
+                json.dump(dumped, f, indent=2)
+            else:
+                yaml.safe_dump(dumped, f, sort_keys=False)
+
+    def _create_validation_structure(self, view: dm.View) -> type[Model]:
+        field_definitions: dict[str, tuple[type, Any]] = {}
```

Comment:
I see that you here want to split between what is a Node (only attributes about node) and what is a Edge (i.e. only references/links) ? This would double number of queries since you are probably quering:
- only `DatatypeProperties` , which fetch only attributes of node
- only `ObjectProperties`, which fetch edges/connections between nodes